Modules/faulthandler.c

```diff
@@ -212,6 +216,81 @@ faulthandler_dump_traceback(int fd, int all_threads,
     reentrant = 0;
 }
 
+#ifdef HAVE_EXECINFO_H
+static void
+faulthandler_stack_dump_impl(int fd)
+{
+#define SIZE 128
+#define ENTRY_SIZE 256
+    void *callstack[SIZE];
+    int frames = backtrace(callstack, SIZE);
+    char **strings = backtrace_symbols(callstack, SIZE);
+    if (strings == NULL)
+    {
+        PUTS(fd, "  <failed to get stack trace>");
+    }
+    else {
+        for (int i = 0; i < frames; ++i)
+        {
+            char entry_str[ENTRY_SIZE];
+            snprintf(entry_str, ENTRY_SIZE, "  %s\n", strings[i]);
+            size_t length = strlen(entry_str) + 1;
+            if (length == ENTRY_SIZE)
+            {
+                /* We exceeded the size, make it look prettier */
+
+                // Add ellipsis to last 3 characters (but before the newline and null terminator)
+                for (int x = 0; x < 3; ++x)
+                    entry_str[ENTRY_SIZE - (x + 3)] = '.';
+
+                entry_str[ENTRY_SIZE - 2] = '\n';
+                entry_str[ENTRY_SIZE - 1] = '\0';
+            }
+            _Py_write_noraise(fd, entry_str, length);
+        }
+
+        if (frames == SIZE)
+        {
+            PUTS(fd, "  <truncated rest of calls>");
+        }
+    }
+#undef ENTRY_SIZE
+#undef SIZE
+}
+#else
+static void
+faulthandler_stack_dump_impl(int fd)
+{
+    PUTS(fd, "  <cannot get C stack on this system>\n");
+}
+#endif
+
+static void
+faulthandler_dump_c_stack_nocheck(int fd)
+{
+    PUTS(fd, "Current thread's C stack trace (most recent call first):\n");
+    faulthandler_stack_dump_impl(fd);
+}
+
+static void
+faulthandler_dump_c_stack(int fd)
+{
+    static volatile int reentrant = 0;
+
+    if (reentrant)
+        return;
```

Comment:
```suggestion
    if (reentrant) {
        return;
    }
```

Reply:
Same here. I want to stay consistent with `faulthandler_dump_traceback`